core/includes/database/database.inc

```diff
@@ -737,12 +737,12 @@ abstract class DatabaseConnection {
       if ($options['throw_exception']) {
         // Add additional debug information.
         if ($query instanceof DatabaseStatementInterface) {
-          $e->query_string = $stmt->getQueryString();
+          $e->errorInfo['query_string'] = $stmt->getQueryString();
```

Comment:
This may be ok, but I was wondering if we could test this approach somehow (unless you have already done this alread). `errorInfo` is a property of `PDOException` and has a very specific usage, which three numeric keys that indicate specific types of errors (see [this](https://www.php.net/manual/en/pdo.errorinfo.php)). I may be over-cautious, but setting additional non-standard keys into that property like `'query_string'` may cause those to be lost at best, or have unintended effects at worst?  I don't really have an alternative here other than extending PDFException to accommodate for two the additional keys `query_string` and `args` - but this may not be necessary if the above has been tested and works.

Or perhaps there are automated tests already that may have tested this change...

Reply:
Sounds good. Resolving this, then.